extension-iap/src/iap_private.cpp

```diff
@@ -113,7 +113,7 @@ void IAP_Queue_Push(IAPCommandQueue* queue, IAPCommand* cmd)
 
     if(queue->m_Commands.Full())
     {
-        queue->m_Commands.OffsetCapacity(2);
+        queue->m_Commands.OffsetCapacity(8);
```

Comment:
I'm afraid this isn't a fix.
It might remedy the symptom, but it isn't the root cause.

This function is pushing a single command, and if the m_Commands array is full, it should suffice to increase the size by 1. The `2` offset is just a tiny optimization to avoid reallocating it for each call to this function.

Instead, we should look into why it crashes.
One thing that comes to mind is the fact that the OffsetCapacity (currently) causes the array to always use a new memory area, so we should be careful with any usage of pointers into this array.
E.g. is there a place where we use this array, that isn't protected by the `queue->m_Mutex`?

Reply:
Sorry, i'm not very strong with c++, i just found problem)

Reply:
We appreciate that you contribute! Let's try to break this down into some very clear steps to reproduce and then check the code for signs of the problem @JCash mentioned. So what are the steps to reproduce?

1. Buy consumable product
2. Call acknowlegde on the product
3. Repeat steps 1 & 2 a number of times (how many?)
4. Shut down the game
5. Open the game again
6. Notice the crash

Reply:
2. yes, here code sample
```
 if provider_id == iap.PROVIDER_ID_GOOGLE then
            iap.acknowledge(store_transaction)
        end
```
3. our game was broken after 4 - 5 times, can't test right now
6. Game crashes every time, but not always with error in logcat

Reply:
Thank you for clarifying. I will try to reproduce this as soon as I have some free time.